lib/benchee/formatter.ex

```diff
@@ -15,57 +15,63 @@ defmodule Benchee.Formatter do
 
   alias Benchee.{Suite, Utility.Parallel}
 
+  @type options :: any
+
   @doc """
   Takes the suite and returns whatever representation the formatter wants to use
   to output that information. It is important that this function **needs to be
   pure** (aka have no side effects) as benchee will run `format/1` functions
   of multiple formatters in parallel. The result will then be passed to
   `write/1`.
   """
-  @callback format(Suite.t()) :: any
+  @callback format(Suite.t(), options) :: any
 
   @doc """
   Takes the return value of `format/1` and then performs some I/O for the user
   to actually see the formatted data (UI, File IO, HTTP, ...)
   """
-  @callback write(any) :: :ok | {:error, String.t()}
+  @callback write(any, options) :: :ok | {:error, String.t()}
 
   @doc """
   Combines `format/1` and `write/1` into a single convenience function that is
   also chainable (as it takes a suite and returns a suite).
   """
-  @callback output(Suite.t()) :: Suite.t()
+  @callback output(Suite.t(), options) :: Suite.t()
```

Comment:
What do you think about getting rid of the `output/2` as a callback - it seems very weird - as far as I understand you never want to override it.

How about defining a regular `Benchee.Formatter.output(formatter, suite, options)` function that would do this? This could allow removing the `use` from `Benchee.Formatter` entirely and keeping it a plain & simple behaviour with no macro magic.

Reply:
Thanks for your feedback @michalmuskala !

:thinking: Basically the `output/2` callback still exists for 2 reasons:

1. Backwards compatibility-ish (as it's now `/2` but I also just had the idea to let the second arg default thereby keeping `output/1`)
2. Convenience access to format + write especially for people who use the more verbose piped interface. Of course they could also just use `Benchee.Formatter.output` then - seems a bit weird.

Not sure, generally I like getting rid off `use` so that's a plus. Maybe it's just hard for me to let go :grin: One thing for me would be that you could no longer say `&Formatter.output/arity` as the formatter would always have to be a parameter

Reply:
@devonestes input on our good ol friend mister `output`? :)

Reply:
'Tis decided then. That requires even more updates then. I'll see how the rest shapes up and include it later in this PR or remove it as part of a separate PR (maybe more pragmatic option). Maybe I'm also overestimating it.

Ideas for names? Right now `Formatter.output` outputs the whole suite with all configured formatters. What should the output method for just one for formatter be called? `output_one` ? :D